presto-main/src/main/java/com/facebook/presto/type/CharParametricType.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeParameter;
+
+import java.util.List;
+
+import static com.facebook.presto.spi.type.CharType.createCharType;
+
+public class CharParametricType
+        implements ParametricType
+{
+    public static final CharParametricType CHAR = new CharParametricType();
+
+    @Override
+    public String getName()
+    {
+        return StandardTypes.CHAR;
+    }
+
+    @Override
+    public Type createType(List<TypeParameter> parameters)
+    {
+        if (parameters.isEmpty()) {
+            return createCharType(1);
```

Comment:
This is diametrically opposite to the behavior of VARCHAR. Why is that? If there is a specific reason, could you leave a comment explaining it? I looked in the spec, but didn't find much.